platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -146,63 +140,52 @@ public class SignedState implements SignedStateInfo {
     /**
      * Keeps track of reservations on this object.
      */
-    private final ReferenceCounter reservations = new ReferenceCounter(this::destroy, this::onReferenceCountException);
+    private final ReferenceCounter reservations =
+            new ReferenceCounter(this::markEligibleForDeletion, this::onReferenceCountException);
 
     /**
      * The signature verifier used to verify signatures.
      */
     private final SignatureVerifier signatureVerifier;
 
+    private final AtomicBoolean eligibleForDeletion = new AtomicBoolean(false);
+
     /**
-     * Instantiate a signed state.
-     *
-     * @param platformContext   the platform context
-     * @param signatureVerifier the signature verifier
-     * @param state             a fast copy of the state resulting from all transactions in consensus order from all
-     *                          events with received rounds up through the round this SignedState represents
-     * @param reason            a short description of why this SignedState is being created. Each location where a
-     *                          SignedState is created should attempt to use a unique reason, as this makes debugging
-     *                          reservation bugs easier.
-     * @param freezeState       specifies whether this state is the last one saved before the freeze
+     * If false, delete signed states on the thread that removes the last reference count. Otherwise, let the background
+     * deletion handler delete the state. This should be true in production environments and false in unit test
```

Comment:
"This should be true in production environments"

This isn't accurate anymore, since there are production environment uses for deletion on the calling thread

Reply:
fixed